phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -78,7 +87,11 @@
     private PhoenixConnection connection;
     private final long maxSize;
     private final ImmutableBytesPtr tempPtr = new ImmutableBytesPtr();
-    private final Map<TableRef, Map<ImmutableBytesPtr,Map<PColumn,byte[]>>> mutations = Maps.newHashMapWithExpectedSize(3); // TODO: Sizing?
+    /**
+     * Use {@link TreeMap} for {@link #mutations} below to enforce a deterministic ordering for easier testing of partial failures.
+     * @see PartialCommitIT
+     */
+    private final Map<TableRef, Map<ImmutableBytesPtr,RowMutationState>> mutations = Maps.newTreeMap(new TableRefComparator());
```

Comment:
They are being sorted here to allow for consistent partial commit verification in tests. Without a predictable sorting here it is impossible to write robust tests for partial commits.